### PR TITLE
Add working rules to project

### PR DESCRIPTION
**Pull request by jacebenson**
this adds working rules to the project
see `/src/srvices/tickets.js`

### CODE REVIEW

File: api/src/services/tickets/tickets.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
 import { db } from 'src/lib/db'
+import * as fs from 'fs'
+//import { runRules } from 'src/lib/rules/insert';
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:3]: remove dead code

File: api/src/services/tickets/tickets.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
+import rules from 'src/rules/tickets/**.{js,ts}'
+console.log(`rules`, rules);
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:5]: remove logging

File: api/src/services/tickets/tickets.js [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+let rulesArr = Object.keys(rules).map((k) => rules[k])//from obj to arr of objs
+rulesArr.sort((a,b) => a.order-b.order );//order rules asc
+//filter out inactive rules//tbd
+console.log(`rulesArr`, rulesArr);
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:9]: remove logging

File: api/src/services/tickets/tickets.js [hunk 4]
```diff
@@ -1,7 +1,16 @@
...
+
 
 export const tickets = () => {
-  return db.ticket.findMany()
+  return db.ticket.findMany({})
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:13]: remove the curly brackets

File: api/src/services/tickets/tickets.js [hunk 5]
```diff
@@ -10,8 +19,27 @@ export const ticket = ({ id }) => {
...
 }
 
-export const createTicket = ({ input }) => {
+export const lastTicket = () => {
+  return db.ticket.findFirst({
+    orderBy: [{
+      number: 'desc'
+    }],
+  })
+}
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:28]: remove this function. Not needed.

File: api/src/services/tickets/tickets.js [hunk 6]
```diff
@@ -1,7 +1,16 @@
 import { db } from 'src/lib/db'
+import * as fs from 'fs'
+//import { runRules } from 'src/lib/rules/insert';
+import rules from 'src/rules/tickets/**.{js,ts}'
+console.log(`rules`, rules);
+let rulesArr = Object.keys(rules).map((k) => rules[k])//from obj to arr of objs
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:6]: (k) => rules[k]
function(k) {
 return rules[k];
 }

File: api/src/services/tickets/tickets.js [hunk 7]
```diff
@@ -1,7 +1,16 @@
...
+rulesArr.sort((a,b) => a.order-b.order );//order rules asc
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:7]: sort( function (a,b) {
    a.order - b.order;
} );

File: api/src/services/tickets/tickets.js [hunk 8]
```diff
@@ -10,8 +19,27 @@ export const ticket = ({ id }) => {
...
+    }],
+  })
+}
+
+export const createTicket = async ({ input }) => {
+  var lastTicket = await db.ticket.findFirst({orderBy: [{number: 'desc'}],});
+  input.number = (parseInt(lastTicket.number,10)+1).toString()
+  var modifiedInput;
+  rulesArr.forEach((rule)=>{
+    console.log('running', rule, input)
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:35]: set up debug value and only run when debugging

File: api/src/services/tickets/tickets.js [hunk 9]
```diff
@@ -10,8 +19,27 @@ export const ticket = ({ id }) => {
...
+    modifiedInput = rule.command(input);
+    console.log('ran', rule, input)
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:37]: setup debug value to only run this when debugging

File: api/src/services/tickets/tickets.js [hunk 10]
```diff
@@ -10,8 +19,27 @@ export const ticket = ({ id }) => {
...
+  })
+}
+
+export const createTicket = async ({ input }) => {
+  var lastTicket = await db.ticket.findFirst({orderBy: [{number: 'desc'}],});
+  input.number = (parseInt(lastTicket.number,10)+1).toString()
+  var modifiedInput;
+  rulesArr.forEach((rule)=>{
+    console.log('running', rule, input)
+    modifiedInput = rule.command(input);
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:36]: can just do rule.command(input)

File: api/src/services/tickets/tickets.js [hunk 11]
```diff
@@ -10,8 +19,27 @@ export const ticket = ({ id }) => {
...
+    console.log('ran', rule, input)
+  })
+  //var modifiedInput = await runRules(input, 'tickets');
+  console.log(`modifiedInput ${JSON.stringify(modifiedInput)}`)
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:40]: remove logging or put it as debug

File: api/src/services/tickets/tickets.js [hunk 12]
```diff
@@ -10,8 +19,27 @@ export const ticket = ({ id }) => {
...
+export const createTicket = async ({ input }) => {
+  var lastTicket = await db.ticket.findFirst({orderBy: [{number: 'desc'}],});
+  input.number = (parseInt(lastTicket.number,10)+1).toString()
+  var modifiedInput;
+  rulesArr.forEach((rule)=>{
+    console.log('running', rule, input)
+    modifiedInput = rule.command(input);
+    console.log('ran', rule, input)
+  })
+  //var modifiedInput = await runRules(input, 'tickets');
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:39]: remove dead code

File: api/src/services/tickets/tickets.js [hunk 13]
```diff
@@ -10,8 +19,27 @@ export const ticket = ({ id }) => {
...
+  console.log(`modifiedInput ${JSON.stringify(modifiedInput)}`)
   return db.ticket.create({
+    //data: modifiedInput,// input,
```
Comment (shiningblue7) [inline api/src/services/tickets/tickets.js:42]: remove dead code

File: package.json [hunk 14]
```diff
@@ -4,7 +4,8 @@
     "packages": [
       "api",
       "web",
-      "packages/*"
+      "packages/*",
+      "fs/promises"
```
Comment (shiningblue7) [inline package.json:8]: remove fs/promises

File: api/src/services/tickets/tickets.js (hunk 1 above)
Comment (jacebenson) [reply api/src/services/tickets/tickets.js:3]: Done

File: api/src/services/tickets/tickets.js (hunk 2 above)
Comment (jacebenson) [reply api/src/services/tickets/tickets.js:5]: Done

File: api/src/services/tickets/tickets.js (hunk 3 above)
Comment (jacebenson) [reply api/src/services/tickets/tickets.js:9]: Done

File: api/src/services/tickets/tickets.js (hunk 5 above)
Comment (jacebenson) [reply api/src/services/tickets/tickets.js:28]: Done

File: api/src/services/tickets/tickets.js (hunk 10 above)
Comment (jacebenson) [reply api/src/services/tickets/tickets.js:36]: Done

File: api/src/services/tickets/tickets.js (hunk 11 above)
Comment (jacebenson) [reply api/src/services/tickets/tickets.js:40]: Removed added backlog item to add debugging class

File: api/src/services/tickets/tickets.js (hunk 12 above)
Comment (jacebenson) [reply api/src/services/tickets/tickets.js:39]: Done

File: api/src/services/tickets/tickets.js (hunk 13 above)
Comment (jacebenson) [reply api/src/services/tickets/tickets.js:42]: Done

**Pull request closed without merging**